linera-execution/src/system.rs

```diff
@@ -355,6 +355,34 @@ impl Recipient {
 #[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Hash, Default, Debug, Serialize, Deserialize)]
 pub struct UserData(pub Option<[u8; 32]>);
 
+impl UserData {
+    pub fn from_option_string(opt_str: Option<String>) -> Self {
+        // Convert the Option<String> to Option<[u8; 32]>
+        let option_array = match opt_str {
+            Some(s) => {
+                // Convert the String to a Vec<u8>
+                let vec = s.into_bytes();
+                if vec.len() <= 32 {
+                    // Create an array from the Vec<u8>
+                    let mut array = [b' '; 32];
+
+                    // Copy bytes from the vector into the array
+                    let len = vec.len().min(32);
+                    array[..len].copy_from_slice(&vec[..len]);
+
+                    Some(array)
+                } else {
+                    None
+                }
+            }
+            None => None,
+        };
+
+        // Return the UserData with the converted Option<[u8; 32]>
+        UserData(option_array)
+    }
+}
```

Comment:
Actually, we shouldn't silently omit the user data just because it's too long. If someone gives too long a string on the command line we should just fail and print an error.
(Sorry, I should have thought of that earlier!)

Reply:
How about this? https://github.com/linera-io/linera-protocol/pull/2369/commits/62030b45242dc43c00ed5c54b48cdb648d9ca93e